src/pages/index.js

```diff
@@ -92,20 +114,73 @@ import { PopupWithImage } from "../components/PopupWithImage";
     `#${addForm.parentElement.id}`,
     handleProfileAddSubmit
   );
-  const userInfo = new UserInfo(profileName, profileDescription);
 
   imagePopup.setEventListeners();
 
+  //preset profile info on load
+  api.getProfileInfo().then((results) => {
+    console.log(results);
+    profileName.textContent = results.name;
+    profileDescription.textContent = results.about;
+    avatarPhoto.src = results.avatar;
+  });
+
   //functions
+  const deletePopup = new confirmationPopup("#confirmation-modal");
+  deletePopup.setEventListeners();
+
+  function addCard(name, description, template, id) {
+
+    const modelCard = new Card(
+      name,
+      description,
+      template,
+      imageClickHandler,
+      (card) => {
+        console.log(card._id);
+
+        // api.deleteCard(card._id, card.deleteCard()); // 
+        deletePopup.open();
+        deletePopup.setSubmitAction(() =>{
+          api.deleteCard(card._id).then(() =>{
+            card.deleteCard();
+            deletePopup.close();
+          })
+        })
+
+      },
+      toggleLikeButton,
+      initialCardsInfo,
+      id
+    );
+
+    galleryList.prepend(modelCard.addCard());
+  }
 
   function createCard(name, description, template) {
-    const modelCard = new Card(name, description, template, imageClickHandler);
-    cardsSection.addItem(modelCard.addCard());
+    api.addCardInfo(name, description, (name, link, id) => {
+      console.log(template);
+      addCard(name, link, template, id);
+    }).then(
+      addPopup.saveButton.textContent = "Saving..."
+    )
   }
+  // editPopup.saveButton.textContent = "Submit";
+  // addPopup.saveButton.textContent = "Submit";
+
+  function handleProfileEditSubmit(inputElements) {
+    // const saveButton = editPopup.querySelector(".modal__save-button");
+    console.log(editPopup)
+    api
+      .setUserInfo(inputElements.name, inputElements.description)
+      .then((result) => {
+        user.setUserInfo({ name: result.name, about: result.about });
+      })
+      .then(
+        editPopup.saveButton.textContent = "Saving..."
+      );
+      editPopup.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

**Please, correct it everywhere**